skillmap/src/lib/workspaceProvider.ts

```diff
@@ -20,10 +23,26 @@ export async function saveProjectAsync(project: pxt.workspace.Project): Promise<
     await ws.saveProjectAsync(project);
 }
 
-export async function getUserStateAsync(): Promise<UserState | undefined> {
+export async function getUserStateAsync(): Promise<UserState> {
     const ws = await getWorkspaceAsync();
 
-    return ws.getUserStateAsync();
+    let userState = await ws.getUserStateAsync();
+    if (!userState) {
+        userState = {
+            id: guidGen(),
+            completedTags: {},
+            mapProgress: {},
+            version: pxt.skillmap.USER_VERSION
+        };
+    }
+    // Read synchronized skillmap state from cloud profile.
+    const skillmapState = await authClient.getSkillmapStateAsync();
+    userState = {
```

Comment:
this section will later do some syncing to patch local state into the fetched cloud state?

Reply:
The patch is happening right here, actually. This is a little bit confusing because I didn't want to change the shape of UserState. Like before, the entire user object is stored in workspace storage, but the `completedTags` and `mapProgress` fields are also sent to the authClient where they're stored locally and synced to cloud if the user is signed in. `authClient` is now the authoritative source for those two fields in the user state, and the copy that is stored here in workspace is entirely ignored. If this is too messy I'm fine to clean it up.

Reply:
ahhh i see! hmm in that case maybe a comment in the reducer or somewhere in the skillmap just stating that the auth client is the source of truth? i think it's fine to have that be the case